boilerplate/openshift/golang-osd-e2e/update

```diff
@@ -21,7 +21,7 @@ OPERATOR_NAME_CAMEL_CASE=${OPERATOR_PROPER_NAME// /}
 
 mkdir -p "${E2E_SUITE_DIRECTORY}"
 
-E2E_SUITE_BUILDER_IMAGE=registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.24-openshift-4.21
+E2E_SUITE_BUILDER_IMAGE=registry.access.redhat.com/ubi9/go-toolset:latest
```

Comment:
Please use the image tag instead of the latest
https://catalog.redhat.com/en/software/containers/rhel9/go-toolset/61df08166d9a1b7b2aab2344#overview